packages/plugins/scripts-runner/src/index.ts

```diff
@@ -0,0 +1,171 @@
+import { eachSeries } from 'async';
+import { promises as fsPromises, stat } from 'fs';
+import path from 'path';
+import { Embark, Callback, Plugins } from 'embark-core';
+import { Logger } from 'embark-logger';
+import { dappPath } from 'embark-utils';
+import Web3 from "web3";
+import { UnsupportedTargetError, ScriptExecutionError } from './error';
+
+export enum ScriptsRunnerCommand {
+  Execute = 'scripts-runner:execute'
+}
+
+export enum ScriptsRunnerEvent {
+  Executed = 'scripts-runner:script:executed'
+}
+
+export interface ScriptDependencies {
+  web3: Web3 | null;
+  contracts: any;
+  logger: Logger;
+}
+
+export interface ExecuteOptions {
+  target: string;
+  dependencies: ScriptDependencies;
+  forceTracking: boolean;
+}
+
+export default class ScriptsRunnerPlugin {
+
+  private _web3: Web3 | null = null;
+
+  constructor(private embark: Embark, private plugins: Plugins) {
+    // TODO: it'd be wonderful if Embark called `registerCommandHandlers()` for us
+    this.registerCommandHandlers();
+  }
+
+  registerCommandHandlers() {
+    this.embark.events.setCommandHandler(ScriptsRunnerCommand.Execute, this.execute.bind(this));
+  }
+
+  private get web3() {
+    return (async () => {
+      if (!this._web3) {
+        const provider = await this.embark.events.request2('blockchain:client:provider', 'ethereum');
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
+
+  private async execute(target: string, forceTracking = false, callback: Callback<any>) {
+    const targetPath = !path.isAbsolute(target) ? dappPath(target) : target;
+    try {
+      const fstat = await fsPromises.stat(targetPath);
+      if (fstat.isDirectory()) {
+        const dependencies = await this.getScriptDependencies();
+        await this.executeAll({ target: targetPath, dependencies, forceTracking }, callback);
+      } else if (fstat.isFile()) {
+        const dependencies = await this.getScriptDependencies();
+        await this.executeSingle({ target: targetPath, dependencies, forceTracking }, callback);
+      } else {
+        callback(new UnsupportedTargetError(fstat));
+      }
+    } catch (e) {
+      callback(e);
+    }
+  }
+
+  private async executeSingle(options: ExecuteOptions, callback: Callback<any>) {
+    const forceTracking = options.forceTracking;
+    const scriptName = path.basename(options.target);
+    const scriptDirectory = path.basename(path.dirname(options.target));
+    const scriptTracked = await this.embark.events.request2('deployment-tracker:script:tracked', { scriptName });
+
+    if (scriptTracked && (scriptDirectory === this.embark.config.embarkConfig.migrations || forceTracking)) {
+      this.embark.logger.info(`.. ${scriptName} already done`);
```

Comment:
Aesthetic comme
```suggestion
      this.embark.logger.info(`${scriptName} already executed`);
```
You can add back the `..` if they served a purpose, but they seemed a bit random